Sources/StreamVideoSwiftUI/CallView/VisibilityThresholdModifier.swift

```diff
@@ -0,0 +1,107 @@
+//
+
+import StreamVideo
+import SwiftUI
+
+/// The modifier designed to dynamically track and respond to the visibility status of a view within its parent
+/// bounds or viewport. It utilises a user-defined visibility threshold, represented as a percentage, to
+/// determine how much of the view should be visible (both vertically and horizontally) before it's considered
+/// "on screen".
+///
+/// When the visibility state of the view changes (i.e., it transitions between being "on screen" and "off screen"),
+/// a callback is triggered to notify the user of this change. This can be particularly useful in scenarios where 
+/// resource management is crucial, such as video playback or dynamic content loading, where actions might
+/// be triggered based on whether a view is currently visible to the user.
+///
+/// By default, the threshold is set to 30%, meaning 30% of the view's dimensions must be within the parent's
+/// bounds for it to be considered visible.
+struct VisibilityThresholdModifier: ViewModifier {
+    /// State to track if the content view is on screen.
+    @State private var isOnScreen = false {
+        didSet {
+            // Check if the visibility state has changed.
+            guard isOnScreen != oldValue else { return }
+            // Notify the caller about the visibility state change.
+            changeHandler(isOnScreen)
+        }
+    }
+
+    /// The bounds of the parent view or viewport.
+    var bounds: CGRect
+    /// The threshold percentage of the view that must be visible.
+    var threshold: CGFloat
+    /// Closure to handle visibility changes.
+    var changeHandler: (Bool) -> Void
+
+    init(in bounds: CGRect,
+         threshold: CGFloat,
+         changeHandler: @escaping (Bool) -> Void
+    ) {
+        self.bounds = bounds
+        self.threshold = threshold
+        self.changeHandler = changeHandler
+    }
+
+    func body(content: Content) -> some View {
+        content
+            .background(
+                GeometryReader { geometry -> Color in
+                    /// Convert the local frame of the content to a global frame.
+                    let geometryInGlobal = geometry.frame(in: .global)
+                    /// Calculate the global minY, maxY, minX, and maxX of the content view.
+                    let minY = geometryInGlobal.minY
+                    let maxY = geometryInGlobal.maxY
+                    let minX = geometryInGlobal.minX
+                    let maxX = geometryInGlobal.maxX
+
+                    /// Calculate required height and width based on visibility threshold.
+                    let requiredHeight = geometry.size.height * threshold
+                    let requiredWidth = geometry.size.width * threshold
+
+                    /// Check if the content view is vertically within the parent's bounds.
+                    let verticalVisible = (minY + requiredHeight < bounds.maxY && minY > bounds.minY) ||
```

Comment:
Can we write some tests for the calculations here?

Reply:
100%! It was my mistake to not add some from the beginning.